MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,11 +1,125 @@
+        imageView.image = step.image
+        textLabel.text = step.text
+        counterLabel.text = step.questionNumber
+    }
+
+    private func show(quiz result: QuizResultViewModel) {
+        
+        let alertModel = AlertModel(title: result.title,
+                                    message: result.text,
+                                    buttonText: result.buttonText,
+                                    completion: { [weak self]_ in
+                                                guard let self = self else {return}
+                                                self.currentQuestionIndex = 0
+                                                self.questionFactory?.requestNextQuestion()
+                                    })
+    
+        alertPresenter?.showResult(AlertModel: alertModel)
+    }
+
+    private func showAnswerResult(isCorrect: Bool) {
+        if isCorrect {
+            correctAnswer += 1
+        }
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? UIColor().ypGreen : UIColor().ypRed
+        yesButton.isEnabled = false
+        noButton.isEnabled = false
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else {return}
+            self.showNextQuestionOrResults()
+            self.yesButton.isEnabled = true
+            self.noButton.isEnabled = true
+        }
+    }
+    
+    private func showNextQuestionOrResults() {
+        if currentQuestionIndex == questionsAmount - 1 {
+            statisticService.store(correct: correctAnswer, total: questionsAmount)
+            
+            let message = "Ваш результат: \(correctAnswer)/\(questionsAmount)\nКоличество сыгранных квизов: \(statisticService.gamesCount)\nРекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(statisticService.bestGame.date.dateTimeString))\nCредняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%"
+            
+            show(quiz: QuizResultViewModel.init(title: "Этот Раунд Окончен!",
+                                                text: message,
+                                                buttonText: "Cыграть еще раз"))
+            imageView.layer.borderWidth = 0
+            correctAnswer = 0
+        } else {
+            currentQuestionIndex += 1
+            imageView.layer.borderWidth = 0
+            questionFactory?.requestNextQuestion()
+        }
+    }
+    
+
     // MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        alertPresenter = AlertPresenter(present: self)
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+        }
+    
+    // MARK: - QuestionFactoryDelegate
+    func didRecieveNextQuestion(question: QuizQuestion?) {
```

Comment:
Чтобы не допускать опечатки, [настройте проверку правописания](https://sarunw.com/posts/spell-checking-in-xcode/)
```suggestion
    func didReceiveNextQuestion(question: QuizQuestion?) {
```